File: app/src/test/java/com/example/kotlinjsonconverter/abson_test/java/examples/_1/PersonTest.java
package com.example.kotlinjsonconverter.abson_test.java.examples._1;

import static org.junit.Assert.assertEquals;

import static co.ab180.abson.serialization.SerializerKt.serialize;

import org.junit.Test;

public class PersonTest {

    @Test
    public void test() {
        Person person = new Person("Alice", 29);
        String json = "{\"Name\": \"Alice\", \"Age\": 29}";

        assertEquals(json, serialize(person));
    }
}

File: app/src/test/java/com/example/kotlinjsonconverter/abson_test/kotlin/codegen_sample/PersonTest.kt
package com.example.kotlinjsonconverter.abson_test.kotlin.codegen_sample

import co.ab180.abson.serialization.serialize
import com.example.kotlinjsonconverter.codegen_sample.Person
import com.example.kotlinjsonconverter.codegen_test.generator.generate_Person
import com.google.gson.Gson
import org.junit.Assert
import org.junit.Test

class PersonTest {

    @Test
    fun test_serialization() {

        val person = generate_Person()
        val personJsonString = serialize(person)
        Assert.assertEquals(person, Gson().fromJson(personJsonString, Person::class.java))
    }
}